### PR TITLE
header & menu for user pages

### PR DESCRIPTION
**Pull request by MohammedAlghazali**
relates #120 
note: I made the header with z-index = 1 and it's background transparent so it can take the background color from the div behind it, to be suitable for our design as it appears in issue #120 

* I made the header as a common component
* I made the menu in this header
* I made an overlay layer beyond the header menu when it opens

---

### CODE REVIEW

File: client/src/components/HeaderWithMenu/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  reverseShow = () => {
+    const { show } = this.state;
+    this.setState({ show: !show });
+  };
+
+  render() {
+    const { show } = this.state;
+    return (
+      <div>
```
Comment (tomduggan85) [inline client/src/components/HeaderWithMenu/index.js:18]: @MohammedAlghazali I think you don't need this div, the `main-header` div is already wrapping everything.

File: client/src/components/UserContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react';
+import './style.css';
+import HeaderWithMenu from '../HeaderWithMenu';
+
+class UserContainer extends Component {
+  state = {};
+
+  render() {
+    return <HeaderWithMenu />;
```
Comment (tomduggan85) [inline client/src/components/UserContainer/index.js:9]: @MohammedAlghazali Are you planning to add more to this component?

File: client/src/components/HeaderWithMenu/index.js (hunk 1 above)
Comment (MohammedAlghazali) [reply client/src/components/HeaderWithMenu/index.js:20]: oh, yes I made something and delete it and forget to remove this

File: client/src/components/UserContainer/index.js (hunk 2 above)
Comment (MohammedAlghazali) [reply client/src/components/UserContainer/index.js]: yes, I will add other components

File: client/src/components/HeaderWithMenu/index.js (hunk 1 above)
Comment (MohammedAlghazali) [reply client/src/components/HeaderWithMenu/index.js:20]: I divide the work to more than pull request

**Pull request closed without merging**